pkg/kube/kube_handler_test.go

```diff
@@ -1088,137 +1093,523 @@ func setClientsetData(cs fakeCombinedClientset, namespaceNames []existingNs, err
-			if (err != nil || tc.expectedReqResolutionError != nil) && !errors.Is(err, tc.expectedReqResolutionError) {
-				t.Fatalf("got: %+v, want: %+v", err, tc.expectedReqResolutionError)
+
+			if got, want := fakeClient.request, tc.httpValidator.Req; !cmp.Equal(want, got, cmpOpts...) {
+				t.Errorf("mismatch (-want +got):\n%s", cmp.Diff(want, got, cmpOpts...))
 			}
-			if tc.expectedReqResolutionError != nil {
-				return
+
+			if got, want := response, tc.expectedResponse; !cmp.Equal(want, got) {
+				t.Errorf("mismatch (-want +got):\n%s", cmp.Diff(want, got))
 			}
-			if appRepo.Spec.Type != "oci" {
+		})
+	}
+}
 
-				req := httpValidator.getReq()
-				reqURLs := []string{}
-				reqURLs = append(reqURLs, req.URL.String())
+type fakeOCIRepo struct {
+	tags     repoTagsList
+	manifest repoManifest
+}
 
-				if !cmp.Equal(tc.expectedURLs, reqURLs) {
-					t.Errorf("Unexpected URLS: %v", cmp.Diff(tc.expectedURLs, reqURLs))
-				}
-				if tc.expectedHeaders != nil && !cmp.Equal(tc.expectedHeaders, cli.(*httpclient.ClientWithDefaults).DefaultHeaders) {
-					t.Errorf("mismatch (-want +got):\n%s", cmp.Diff(tc.expectedHeaders, cli.(*httpclient.ClientWithDefaults).DefaultHeaders))
-				}
-			} else {
-				httpValidatorAppRepo := httpValidator.getAppRepo()
-				if httpValidatorAppRepo != appRepo {
-					t.Errorf("Unexpected Repos: %v", cmp.Diff(appRepo, httpValidatorAppRepo))
-				}
+// makeTestOCIServer returns a small test double for an OCI server that handles requests
+// for tags/list and manifest only.
+func makeTestOCIServer(t *testing.T, registryName string, repos map[string]fakeOCIRepo, requiredAuthHeader string) *httptest.Server {
+	// Define a map of valid request/responses based on the fake repos passed in.
+	responses := map[string]string{}
+	for repoName, repo := range repos {
+		tags, err := json.Marshal(repo.tags)
+		if err != nil {
+			t.Fatalf("%+v", err)
+		}
+		responses[path.Join("/v2", registryName, repoName, "tags", "list")] = string(tags)
+
+		manifest, err := json.Marshal(repo.manifest)
+		if err != nil {
+			t.Fatalf("%+v", err)
+		}
+
+		if len(repo.tags.Tags) > 0 {
+			responses[path.Join("/v2", registryName, repoName, "manifests", repo.tags.Tags[0])] = string(manifest)
+		}
+	}
+
+	// Return a test server that responds with these canned responses only.
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Required authorization when set.
+		authHeader := r.Header.Get("Authorization")
+		if authHeader != requiredAuthHeader {
+			w.WriteHeader(401)
+			w.Write([]byte("{}"))
+		}
+		if response, ok := responses[r.URL.Path]; !ok {
+			w.WriteHeader(404)
+			w.Write([]byte("{}"))
+		} else {
+			w.Write([]byte(response))
+		}
+	}))
+}
+
+func TestOCIValidate(t *testing.T) {
+	registryName := "bitnami"
+	testCases := []struct {
+		name             string
+		repos            map[string]fakeOCIRepo
+		validator        HelmOCIValidator
+		expectedResponse *ValidationResponse
```

Comment:
With the test double, we can now unit-test the specific OCI Validate() method here, as well as the general handler (including authn) below.